disco_aws_automation/disco_elastigroup.py

```diff
@@ -0,0 +1,242 @@
+"""Contains DiscoElastigroup class that orchestrates AWS Spotinst Elastigruops"""
+import logging
+import time
+import json
+from os.path import expanduser
+from base64 import b64encode
+import requests
+
+logger = logging.getLogger(__name__)
+
+SPOTINST_API = 'https://api.spotinst.io/aws/ec2/group/'
+
+
+class DiscoElastigroup(object):
+    """Class orchestrating elastigroups"""
+
+    def __init__(self, environment_name, token=None, session=None):
+        self.environment_name = environment_name
+        self._token = token or None
+        self._session = session or None
+        self.account_id = '646102706174'
+
+    @property
+    def token(self):
+        """
+        Returns spotinst auth token from JSON file in ~/.aws/spotinst_api_token
+
+        File format example:
+
+        {
+          "name": "user_ampli
+          "token": "f7e6c5abb51bb04fcaa411b7b70cce414c821bf719f7db0679b296e588630515"
+        }
+        """
+        token_file = json.load(open(expanduser('~') + '/.aws/spotinst_api_token'))
+        if token_file:
+            self._token = token_file['token']
+        return self._token
+
+    @property
+    def session(self):
+        """Lazily create session object"""
+        if not self._session:
+            self._session = requests.Session()
+
+        # Insert auth token in header
+        self._session.headers.update(
+            {
+                "Content-Type": "application/json",
+                "Authorization": "Bearer {}".format(self.token)
+            }
+        )
+        return self._session
+
+    def get_new_groupname(self, hostclass):
+        """Returns a new elastigroup name when given a hostclass"""
+        return self.environment_name + '_' + hostclass + "_" + str(int(time.time()))
+
+    def _filter_by_environment(self, groups):
+        """Filters elastigroups by environment"""
+        return [
+            group for group in groups
+            if group['name'].startswith("{0}_".format(self.environment_name))
+        ]
+
+    def get_hostclass(self, group_name):
+        """Returns the hostclass when given an elastigroup name"""
+        return group_name.split('_')[1]
+
+    def get_existing_groups(self, hostclass=None, group_name=None):
+        """
+        Returns all elastigroups for a given hostclass, sorted by most recent creation. If no
+        elastigroup can be found, returns an empty list.
+        """
+        try:
+            groups = self.session.get(SPOTINST_API).json()['response']['items']
```

Comment:
Should the code handle Error response from spotInst :
[SpotInst Error](https://spotinst.atlassian.net/wiki/display/API/API+Semantics#APISemantics-EM)